pkg/ast/ast.go

```diff
@@ -202,6 +202,62 @@ func (d *Document) DeleteRootNode(node Node) {
 	}
 }
 
+type ParentType struct {
+	Ref       int
+	ValueRefs []int
+}
+
+func removeNodeIgnoringOrder(nodes []Node, indexToRemove int) []Node {
+	lastIndex := len(nodes) - 1
+	nodes[indexToRemove] = nodes[lastIndex]
+	return nodes[:lastIndex]
+}
+
+func (d *Document) mergeDuplicateParentTypes(rootNodeRef int, kind NodeKind, set map[string]*ParentType) {
```

Comment:
I'm thinking I could make this more generic by using an interface or something like that. Thoughts?

Reply:
With the interface, we would be able to remove duplicate unions and enums in the same visitor if we wanted. If this is not wanted, the interface is no longer necessary.